datafusion/core/src/physical_optimizer/enforce_distribution.rs

```diff
@@ -1850,8 +1915,12 @@ pub(crate) mod tests {
             config.optimizer.repartition_file_min_size = $REPARTITION_FILE_MIN_SIZE;
             config.optimizer.prefer_existing_sort = $PREFER_EXISTING_SORT;
             config.optimizer.prefer_existing_union = $PREFER_EXISTING_UNION;
-            // Use a small batch size, to trigger RoundRobin in tests
-            config.execution.batch_size = 1;
```

Comment:
Note the removal of the forced insertion of round robin repartitioning. This was preventing our coalesce insertion (as seen in our real world reproducer).